.travis.yml

```diff
@@ -1,32 +1,32 @@
-language: generic
+language: node_js
+node_js:
+  - "10"
 
 env:
   - OCAML=4.02.3
   - OCAML=4.02.3+buckle-master
+  - OCAML=4.02.3+buckle-master ESY_BUILD=YES ESY__CACHE=/home/travis/.esy
   - OCAML=4.03.0
   - OCAML=4.04.2
   - OCAML=4.05.0
   - OCAML=4.06.1
   - OCAML=4.07.0
 
-before_script:
-  - sudo add-apt-repository -y ppa:robert7/tidy-html5
-  - sudo apt-get update
-  - sudo apt-get install tidy
-  - wget https://github.com/ocaml/opam/releases/download/2.0.0/opam-2.0.0-x86_64-linux
-  - sudo mv opam-2.0.0-x86_64-linux /usr/local/bin/opam
-  - sudo chmod a+x /usr/local/bin/opam
-  - "opam init -y --compiler=$OCAML --disable-sandboxing"
+cache:
+  directories:
+  - /home/travis/.opam
+  - /home/travis/.esy
+  - /home/travis/.nvm
+  - ./_build/default
+
+install:
+  - ./.ci/install.sh
```

Comment:
I recommend not breaking this up into files. If you'd like to move it to a shell script (for conditionals?), I suggest to create one shell script, and do things there. Perhaps one for building and one for installing, if caching happens between those two steps.

With many scripts, and some commands in the `.travis.yml` file, people have to look through a lot of little files to understand what is going on.

Reply:
@aantron a small problem with this approach is that the `eval $(opam env)` call has to be made from an individual line in the `.travis.yml` file to affect the rest of the build.

`eval` from within scripts only affects the current environment, not it's parent.

We can bundle the `install.sh` and `install-esy.sh`, but we'd need a separate script that is run after the `eval` to be able to output the `ocaml/opam -[-]version` commands.

```yaml
install:
  - ./.ci/install.sh
  - eval $(opam env)
  - ./.ci/print-versions.sh
```

Let me know what you think.